R/impute.R

```diff
@@ -4,53 +4,59 @@
+#' information.
 #'
-#' @param strategies a named list of functions. Defines the imputation functions to be used. The names
-#' of the list should mirror the values specified in `method` column of `data_ice`.
+#' @param strategies a named list of functions. Defines the imputation functions to be used.
+#' The names of the list should mirror the values specified in `strategy` column of `data_ice`.
 #' Default = `getStrategies()`. See [getStrategies()] for more details.
 #'
 #' @details
 #'
-#' `impute` works by taking the model parameter estimates generated by [draws()] and using them to calculate
-#' the conditional distribution for the outcome for each subject separately. These distributions are then augmented
-#' based upon the date of the intercurrent event (ICE), the selected strategy and the chosen reference group. Each imputed dataset is
-#' then constructed based upon the method as specified in the call to [draws()] with 1 dataset being generated per
-#' set of parameter estimates provided by [draws()].
-#'
-#' - Bayes & Approximate Bayes: each imputed dataset consists of 1 row per subject & visit from the original
-#' dataset with missing values imputed by taking a single random sample from the augmented outcome distribution.
-#'
-#' - Conditional Mean: each imputed dataset consists of 1 row per subject & visit from the bootstrapped or Jackknife
-#' dataset that was used to generate the corresponding parameter estimates in [draws()].
-#' Missing values are imputed by using the mean of the augmented outcome distribution.
-#'
-#' For conditional mean and approximate Bayes the first imputed dataset will be the original dataset imputed
-#' using the ML parameter estimates of the
-#' MMRM model fitted to the original dataset. See TODO (reference paper for methods) for full details.
-#'
-#'
-#' `update_strategy` can be used to update the imputation method that was originally set via
-#' the `data_ice` option in `draws`.
-#' The `update_strategy` dataframe must contain two columns, one for the subject ID and another for the imputation method,
-#' whose names are the same as
-#' those defined in `vars` argument as specified in the call to `draws`.
-#' You can only update the imputation method, that is to say that specifying
-#' a visit column has no effect here. If you wish to change the ICE visit you will need to re-run `draws`.
-#' A key limitation is that you can only switch between a MAR and a non-MAR method (or vice versa)
-#' if that subject has no non-missing post ICE data.
-#' As an example, if a subject had their ICE on "Visit 2" but still had values in the dataset for
-#' "Visit 3" then the function will throw and error if you try to switch them from MAR to a non-MAR method or
-#' from a non-MAR method to MAR.
+#' `impute()` works by taking the model parameter estimates, as generated by [draws()], and using
+#' them to calculate the conditional distribution for the outcome for each subject separately.
```

Comment:
"conditional" -> "marginal"

Reply:
But we calculate the conditional distribution for each subject based on their covariates ? I.e. we take the model and fit it using the patients baseline characteristics ? Thats a conditional distribution no ?

Reply:
Yes, you're right, the naming is just a bit confusing since we then condition also on the observed dat after. How we usually tell the story, is that we first build the "marginal imputation distribution" for each patient, and then we build the conditional distribution conditioning on observed outcome values. But we can also keep this naming you are having, just mentioning after that we additionally condition on the observed data.

Reply:
Perhaps I should be more explicit with my phrasing then perhaps:

```
#' `impute()` works by taking the model parameter estimates, as generated by [draws()], and using
#' them to calculate the distribution of the outcome variable separately for each subject by conditioning on their covariates.
```

Reply:
sounds good!

Reply:
done